core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -336,7 +336,10 @@ private static int parseVersion(String metadataLocation) {
     int versionEnd = metadataLocation.indexOf('-', versionStart);
     try {
       return Integer.valueOf(metadataLocation.substring(versionStart, versionEnd));
-    } catch (NumberFormatException e) {
+    } catch (NumberFormatException | StringIndexOutOfBoundsException e) {
```

Comment:
Why not check `versionEnd`? If `versionEnd` is < 0, then return -1. No need to catch an exception.